db/cryptodata/client.go

```diff
@@ -0,0 +1,90 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package cryptodata
+
+import (
+	"crypto/rsa"
+	"github.com/ligato/cn-infra/db/keyval"
+	"errors"
+	"crypto/rand"
+	"io"
+	"encoding/base64"
+	"hash"
+	"crypto/sha256"
+)
+
+// ClientConfig is result of converting Config.PrivateKeyFile to PrivateKey
+type ClientConfig struct {
+	// Private key is used to decrypt encrypted keys while reading them from store
+	PrivateKeys []*rsa.PrivateKey
+	// Reader used for encrypting/decrypting
+	Reader io.Reader
+	// Hash function used for hashing while encrypting
+	Hash hash.Hash
+}
+
+// Client handles encrypting/decrypting and wrapping data
+type Client struct {
+	ClientConfig
+}
+
+// NewClient creates new client from provided config and reader
+func NewClient(clientConfig ClientConfig) (client *Client) {
+	client = &Client{
+		ClientConfig: clientConfig,
+	}
+
+	// If reader is nil use default rand.Reader
+	if client.Reader == nil {
+		client.Reader = rand.Reader
+	}
+
+	// If hash is nil use default sha256
+	if client.Hash == nil {
+		client.Hash = sha256.New()
+	}
+
+	return
+}
+
+// EncryptData encrypts input data using provided public key
+func (client *Client) EncryptData(inData []byte, pub *rsa.PublicKey) (data []byte, err error) {
+	data, err = rsa.EncryptOAEP(client.Hash, client.Reader, pub, inData, nil)
+	data = []byte(base64.URLEncoding.EncodeToString(data))
```

Comment:
Actually, this might be required in json. However it's gonna be unnecessary for types that work with `[]byte`, since it will unnecessarily make it larger. It seems to me it should be used only for the json decrypter.